### PR TITLE
docs: add docs for ratelimit invert match headers

### LINKED ISSUE #2193: AND NOT style matches for RateLimiting
*Description*:
>Describe the desired behavior, what scenario it enables and how it
would be used.

Consider this config
```
cat <<EOF | kubectl apply -f -
apiVersion: gateway.envoyproxy.io/v1alpha1
kind: SecurityPolicy
metadata:
  name: jwt-backend
spec:
  targetRef:
    group: gateway.networking.k8s.io
    kind: HTTPRoute
    name: backend
  jwt:
    providers:
    - name: example
      remoteJWKS:
        uri: https://test.eu.auth0.com/.well-known/jwks.json
      claimToHeaders:
      - claim: email
        header: x-user-email
---
apiVersion: gateway.envoyproxy.io/v1alpha1
kind: BackendTrafficPolicy 
metadata:
  name: policy-httproute
spec:
  targetRef:
    group: gateway.networking.k8s.io
    kind: HTTPRoute
    name: backend
    namespace: envoy
  rateLimit:
    type: Global
    global:
      rules:
      - clientSelectors:
        - headers:
          - name: x-user-email
            value: myemail@gmail.com
        limit:
          requests: 10
          unit: Minute
    - clientSelectors:
      - headers:
        - type: Distinct
          name: x-claim-email
      limit:
        requests: 5
        unit: Minute
---
apiVersion: gateway.networking.k8s.io/v1
kind: HTTPRoute
metadata:
  name: backend
spec:
  parentRefs:
    - name: eg
      namespace: envoy
  hostnames:
    - "www.example.com"
  rules:
    - backendRefs:
        - group: ""
          kind: Service
          name: backend
          port: 3000
          weight: 1
      matches:
        - path:
            type: PathPrefix
            value: /
EOF
```

I want to allow [myemail@gmail.com](mailto:myemail@gmail.com) user to 10 req/min but all other user 5 req/min.
But with this config [myemail@gmail.com](mailto:myemail@gmail.com) user is getting 429 after 5 requests.

Relates to https://envoyproxy.slack.com/archives/C03E6NHLESV/p1699784524023299

[optional *Relevant Links*:]
>Any extra documentation required to understand the issue.

### PR DESCRIPTION
**Pull request by rudrakhp**
**What type of PR is this?**

docs: add docs for ratelimit invert match headers
api: remove `notImplementedHide` for `Invert` feature
e2e: add a test for invert header match in local ratelimit case

**What this PR does / why we need it**:

Docs for feature implemented in #4286 

**Which issue(s) this PR fixes**:

Fixes #2193

### CODE REVIEW

**Comment by rudrakhp** [comment]
Added docs manually but `gen-check` seems to be reverting those changes. Am I missing something here?

**Comment by zirain** [comment]
you should update docs under `latest` folder, that's the doc for the next version.

docs alias to current version, which means `v1.1` now.

File: site/content/en/latest/tasks/traffic/local-rate-limit.md [hunk 1]
```diff
@@ -245,6 +245,188 @@ server: envoy
...
+  - group: gateway.networking.k8s.io
+    kind: HTTPRoute
+    name: http-ratelimit
+  rateLimit:
+    type: Local
+    local:
+      rules:
+      - clientSelectors:
+        - headers:
+          - name: x-user-id
```
Comment (zhaohuabing) [inline site/content/en/latest/tasks/traffic/local-rate-limit.md:273]: NIt: typically, user's role is tied to a specific user id. Could we change the first header in this example to something like "x-user-org"?

like:
```
   - name: x-user-org
    value: foo
    - name: x-user-role
    ....
```

**Review by zhaohuabing — DISMISSED** [review]
LGTM, the example could be improved but it's not blocking.

File: site/content/en/latest/tasks/traffic/local-rate-limit.md (hunk 1 above)
Comment (rudrakhp) [reply site/content/en/latest/tasks/traffic/local-rate-limit.md:273]: Agree, changed the example to not rate limit if `x-ord-id` is set to `test`.

**Review by zhaohuabing — APPROVED** [review]
LGTM thanks!

**Review by shawnh2 — APPROVED** [review]
LGTM

**Review by zhaohuabing — APPROVED** [review]

**Pull request merged**